scripts/lcc_to_class_subclass.py

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env python3
+
+from __future__ import print_function
+
+import json
+import os
+import string
+
+import requests  # python3 -m pip install requests
+
+# from typing import Tuple
+
+here = os.path.dirname(os.path.abspath(__file__))
+
+# Load two Python dicts with Library of Congress Classification conversion data
+with open(os.path.join(here, "lc_classifiers_letters_only.json")) as in_file:
+    lcc_letters_only = json.load(in_file)
+with open(os.path.join(here, "lc_classifiers_letters_and_numbers.json")) as in_file:
+    lcc_letters_and_numbers = json.load(in_file)
+out_file = None
+
+
+def get_ol_book_info(olid="OL26617202M"):
+    """
+    def get_ol_book_info(olid: str = "OL26617202M") -> str:
+
+    >>> get_ol_book_info()  # doctest: +ELLIPSIS
+    {'olid:OL26617202M': ...
+    """
+    url = "https://openlibrary.org/api/books?jscmd=details&format=json&bibkeys=olid:"
+    return requests.get(url + olid).json()
+
+
+def olid_to_lc_classifications(olid="OL1025841M"):
+    """
+    def olid_to_lc_classifications(olid: str = "OL1025841M") -> list:
+
+    >>> olid_to_lc_classifications()
+    ['HB1951 .R64 1995']
+    """
+    return get_ol_book_info(olid)["olid:" + olid]["details"]["lc_classifications"]
+
+
+def parse_lcc(lcc):
+    """
+    def parse_lcc(lcc: str) -> Tuple[str, int]:
+
+    >>> parse_lcc("HB1951 .R64 1995")
+    ('HB', 1951)
+    >>> parse_lcc("OL1025841M")
+    ('HB', 1951)
+    >>> parse_lcc("DP402.C8 O46 1995")
+    ('DP', 402)
+    >>> parse_lcc("CS879 .R3 1995")
+    ('CS', 879)
+    """
+    lcc = lcc.strip().upper()
+    if lcc.startswith("OL"):  # User entered an OL number so convert it to LCC
+        return parse_lcc(olid_to_lc_classifications(lcc)[0])
+    lcc = lcc.replace(" ", "").replace(".", " ").replace("/", " ").split()[0].upper()
+    chars = ""
+    for i, char in enumerate(lcc):
+        if char in string.ascii_uppercase:
+            chars += char
+        else:
+            return chars, int(lcc[i:])
+    return chars, 0
+
+
+def lcc_to_classification(lcc):
```

Comment:
Maybe this name might be better?

```suggestion
def get_lcc_classes(lcc):
```

Reply:
@cdrini the MLCs do have classes in parentheses, but I could see leaving it out for now - as it's extra coding. They just don't have subclasses, but just classes is something that should be included (maybe one day).

Reply:
The name change does look a lot cleaner and more readable :), but I'll leave it to the code writers to make the final decision.